### DISCUSSION TITLE
Getting 404 error in my client side when I deploy to vercel

### DISCUSSION THREAD

**Discussion by dantakelly** [vercel/community#7335] [Help]
### Summary

Hey guys, Kelly here. I'm developing a chat app called Social-Flow. I've deployed both the client and server sides of this app to Vercel. The server side is built with Node/Express, and the client side uses Next.js. I've configured the server side using vercel.json, and it's working fine. However, the client side is giving me a 404 error with the code: NOT FOUND.

After some research, I found that I need a vercel.json file for the client side as well. Initially, I added this configuration based on my research:

{
  "routes": [
    {
      "src": "/[^.]+",
      "dest": "/"
    }
  ]
}

But Vercel support suggested I use the following instead:

{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/index"
    }
  ]
}

Despite these changes, I'm still getting the 404 error. Does anyone know what I should do? The issue is with the client side when I launch it on Vercel, not the server side. I made sure that the root directory is set to the client side when deploying to Vercel.

### Steps to Reproduce

Code in the client side verce.json file

code from research 

{
  "routes": [
    {
      "src": "/[^.]+",
      "dest": "/"
    }
  ]
}

code from vercel support ai 

{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/index"
    }
  ]
}

**Comment by dantakelly** [comment]
I then added this code that now shows the client side but the css is not in only the html - {
    "version": 2,
    "builds": [
      {
        "src": "package.json",
        "use": "@vercel/next"
      }
    ],
    "routes": [
      {
        "src": "/(.*)",
        "dest": "/"
      }
    ]
  }
  Any help?

**Comment by pawlean** [comment]
Hi @dantakelly!

Have you tried going onto your project settings? Under "Root Directory", you may specify the right destination.

**Comment by dantakelly** [comment]
I got it to work! I had to redeploy though.

Thank you!

**Reply by pawlean** [reply]
Thanks for coming back here! What were the steps that you took?

**Reply by dantakelly** [reply]
I navigated to the root directory and modified the settings to client-side deployment. However, the initial deployment attempt was unsuccessful, so I repeated the process to ensure a successful outcome.

[a comment by an automated account is not shown]
